### ISSUE TITLE
Make syncpack configurable from the 'config' option in package.json

### ISSUE THREAD

**Issue by bombillazo** [JamieMason/syncpack#86]
## Description

Currently, when configuring `syncpack` from package.json, it is only configurable with the `syncpack` option directly in the root of the package.json file. For compatibility with the [config](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#config) option, it would be great if `syncpack` also checked for the config inside the config option.

## Suggested Solution

Have `syncpack` check for the configuration inside `config`:

```json
{
  ...
  "config" : {
    "syncpack" : {
      // syncpack configuration
    }
}
```

**Comment by JamieMason** [comment]
Hey Hector, this is all handled by https://github.com/davidtheclark/cosmiconfig – not sure yet but maybe it's some option we need to pass or it's something they might add as a new feature on request.

**Comment by bombillazo** [comment]
Hey, thanks for the response, here's the prop you're looking for: [packageProp](https://github.com/davidtheclark/cosmiconfig#packageprop) : ['configs', 'syncpack']

Looks like it only allows one value for package.json, it would've been nice if multiple values were could be configured.

**Comment by JamieMason** [comment]
Had a quick look at this just now @bombillazo, what it would need is for syncpack to run cosmiconfig a 2nd time if `result` is `null` the first time. 

https://github.com/JamieMason/syncpack/blob/2d9516f02b579ecb9ebc1c632ae4db997d796312/src/lib/disk.ts#L31

I'll lower the priority on this rather than close it, as I can see why it would be desirable. But a relative minority of users would use it, while all users which don't use the default config file location would get a performance penalty.

Thanks for raising, let's look again in a while.

**Comment by bombillazo** [comment]
Sure, one recommendation I could give is to directly read the package.json file and see which keys exist before running  cosmiconfig:

```ts
var pjson = require('./package.json');
const hasConfigOption = !!config.syncpack
```

**Comment by JamieMason** [comment]
Yeah, good shout 👍🏻

**Comment by JamieMason** [comment]
Released in [8.3.8](https://github.com/JamieMason/syncpack/releases/tag/8.3.8), thanks a lot.

**Comment by bombillazo** [comment]
Thank you!
